zynautoconnect/zynthian_autoconnect.py

```diff
@@ -513,20 +512,41 @@ def audio_autoconnect(force=False):
 				continue
 
 			#Get Root Layer Input ports ...
-			rl_in=jclient.get_ports(rl.get_jackname(), is_input=True, is_audio=True)
+			rl_in=get_input_ports(rl)
+
+			if len(rl.audio_in) == 0 and len(rl_in) > 0:
+				#Standard routing: Connect System Capture to Root Layer ports
+				for c in range(0, max(2, len(system_capture))):
+					if (len(rl_in) > c):
+						rl.audio_in[system_capture[c].name] = c
+					else:
+						rl.audio_in[system_capture[c].name] = 0
+				logging.info("Created standard routing: {}".format(rl.audio_in))
 
-			#Connect System Capture to Root Layer ports
-			if len(rl_in)>0:
-				if len(rl_in)==1:
-					rl_in.append(rl_in[0])
+			# First, disconnect all capture ports from all inputs
+			for cap in system_capture:
+				for c in range(0, len(rl_in)):
+					try:
+						jclient.disconnect(cap, rl_in[c])
+					except:
+						pass
```

Comment:
Ups! It can't be done like that. For avoiding "clicks" we have to disconnect the inputs that are not connected, but never disconnect a currently connected pair. If not, every time you add/change an effect, everything get disconnected and reconnected, generating lot of noises, especially if there is sound flowing thru the chain.

Reply:
I am aware, and was a little afraid of that would be needed :) That's why I called it WIP, and didn't merge it anywhere yet.

We can totally do it; however, on the hifiberry I'm testing with, there were no huge clicks while playing around.